packages/shell-api/src/collection.ts

```diff
@@ -963,6 +963,29 @@ export default class Collection extends ShellApiWithMongoClass {
     );
   }
 
+  /**
+   * Compacts structured encryption data.
+   *
+   * @return {Promise}
+   */
+  @returnsPromise
+  @apiVersions([])
+  async compactStructuredEncryptionData(): Promise<Document> {
+    // @ts-expect-error waiting for driver release
+    const encryptedFieldsMap = this._mongo._fleOptions?.encryptedFieldsMap;
+    const encryptedFields: Document | undefined = encryptedFieldsMap?.[`${this._database._name}.${ this._name}`];
+
+    if (!encryptedFields) {
```

Comment:
This check is too strict -- it doesn't matter whether the collection is present in the `encryptedFieldsMap`, it matters whether it's an FLE2 collection on the server side.